docs/semgrep-code/java.md

```diff
@@ -0,0 +1,80 @@
+---
+slug: java
+append_help_link: true
+title: Semgrep Pro in Java
+hide_title: true
+description: How Semgrep Pro Engine can increase true positives and reduce false positives in the Java language.
+tags:
+  - tk
+---
+
+# Semgrep Pro performance in Java
+
+<!-- False positive reduction in Java through Semgrep Pro -->
+
+This document explains how Semgrep Pro Engine detects true positives and reduces false positives in Java.
+
+Additionally, it provides several simple rule examples to illustrate the concepts and how you can make use of these Semgrep features when writing your own rules.
+
+## Language features prevent injection through boolean and integer types 
+
+Strong typing in Java, combined with its compile-time and runtime checks, reduces the likelihood that an integer or boolean input will be exploited to perform injection-style attacks. Semgrep Pro can reduce false positives by leveraging these checks.
+
+The OSS Engine matches based on patterns, which can result in false positivies (FPs), but the Pro Engine can detect boolean and integer values and mark these as untainted, or safe, eliminating FPs.
+
+### Example: `int-bool-untainted`
+
+<iframe title="tk" src="https://semgrep.dev/embed/editor?snippet=Ab0p8" width="100%" height="432px" frameBorder="0"></iframe>
```

Comment:
`interfile: true` should be inside `options:`, using it inside `metadata:` is deprecated.

Reply:
Corrected, thank you.